src/lib/grammar.js

```diff
@@ -129,12 +129,13 @@ function buildRegex(map, keyEx, valEx) {
         '(?<' + keyEx + '>',
             keys,
         ')',
-        // value is optional
         '(?:\\(',
             '(?<' + valEx + '>',
                 GRAMMAR.VALUES,
             ')',
-        '\\))?',
+        '\\))',
+        // value is optional only for helper
+        keyEx === 'helper' ? '?' : ''
```

Comment:
I like the idea of adding a flag to tell when param is required or not, but I wouldn't touch the "matcher" part. I prefer adding another key to the object saying the param is optional.